docs/local_assembly.tex

```diff
@@ -0,0 +1,77 @@
+\documentclass[nofootinbib,amssymb,amsmath]{revtex4}
+\usepackage{mathtools}
+\usepackage{amsthm}
+\usepackage{algorithm}
+\usepackage{algpseudocode}
+\usepackage{lmodern}
+\usepackage{graphicx}
+\usepackage{color}
+
+%Put an averaged random variable between brackets
+\newcommand{\ave}[1]{\left\langle #1 \right\rangle}
+\newcommand{\HC}{\texttt{HaplotypeCaller}}
+\newcommand{\Mutect}{\texttt{Mutect}}
+\newcommand{\code}[1]{\texttt{#1}}
+
+\newtheorem{lemma}{Lemma}
+\newtheorem{corollary}{Corollary}
+
+\def\SL#1{{\color [rgb]{0,0,0.8} [SL: #1]}}
+\def\DB#1{{\color [rgb]{0,0.8,0} [DB: #1]}}
+
+\begin{document}
+
+\title{Local Assembly in HaplotypeCaller and Mutect}
+\author{David Benjamin\footnote{The author took no part in development of the methods described below -- credit belongs to several others on the GATK team. }}
+\email{davidben@broadinstitute.org}
+\affiliation{Broad Institute, 75 Ames Street, Cambridge, MA 02142}
+
+\date{\today}
+
+\begin{abstract}
+The GATK tool \HC~ and \Mutect~ assemble reads aligned within a window of several hundred base pairs into a de Bruijn graph of local variation.  Local haplotypes correspond to paths in this de Bruijn graph, and the downstream likelihood calculations in both tools involve aligning reads to these haplotypes.  Here we describe how this de Bruijn graph is generated.
+\end{abstract}
+
+\maketitle
+
+\section{Correcting Reads} \label{correcting-reads}
+Read error correction is turned off by default in \HC~ and \Mutect~ and we are not familiar with it.  The idea is as follows: large kmers are good because they contain more phasing information and therefore yield a simpler de Bruijn graph.  For example, two SNVs 20 bases apart with sequenced with error-free reads yield a de Bruijn graph with $2 \times 2 = 4$ paths if $k = 10$ but only $2$ paths if $k = 40$, because the latter spans the phased SNVs.  The drawback is that larger kmers are more liable to be lost due to sequencing error simply because they contain more bases.  The idea of read error correction is to rescue kmers from the occasional error.
```

Comment:
@ronlevine Thanks so much for looking at this! (Not that I'm conscripting you to review, but I appreciate any participation).  Long kmers yield two paths because there is a bubble one side of which is the reference and one side of which contains the two phased variants (or one side contains one variant if they are anti-phased).  With short kmers you get a bubble for the first variant, followed by a stretch of ref-only, followed by a second bubble.  I can clarify this in the docs.  Or maybe even make a figure.